netx/bytecounting/bytecounting.go

```diff
@@ -0,0 +1,56 @@
+// Package bytecounting adds bytecounting to measurable objects.
+package bytecounting
+
+import (
+	"context"
+	"net"
+
+	"github.com/ooni/probe-engine/atomicx"
+	"github.com/ooni/probe-engine/netx/measurable"
+)
+
+// TODO(bassosimone): we can estimate the bytes required by LookupHost
+// as well using an algorithm similar to MK's one.
+
+// Counter adds byte counting to measurable operations.
+type Counter struct {
+	measurable.Operations
+	BytesSent *atomicx.Int64
+	BytesRecv *atomicx.Int64
+}
+
+// NewCounter creates a new instance of counter.
+func NewCounter(ops measurable.Operations) *Counter {
+	return &Counter{
+		Operations: ops,
+		BytesRecv:  atomicx.NewInt64(),
+		BytesSent:  atomicx.NewInt64(),
+	}
+}
+
+// DialContext implements measurable.Connector.DialContext
+func (c *Counter) DialContext(
+	ctx context.Context, network, address string) (net.Conn, error) {
+	conn, err := c.Operations.DialContext(ctx, network, address)
+	if err != nil {
+		return nil, err
+	}
+	return &connWrapper{Conn: conn, Counter: c}, err
+}
```

Comment:
The specific problem of this implementation is that the byte counter is also a dialer, therefore we cannot separate the byte counting and the dialing. However, we sometimes want to account for session and sometimes for both session and experiment. Therefore, this implementation here is not actually meeting our objectives.